src/test/java/org/mockitousage/debugging/VerificationListenerCallBackTest.java

```diff
@@ -0,0 +1,123 @@
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.*;
+
+public class VerificationListenerCallBackTest {
+    public static final String invocationWanted = "doSomething";
+
+    @Test
+    public void should_call_single_listener_on_verify() {
+        //given
+        RememberingListener listener = new RememberingListener();
+        ThreadSafeMockingProgress.mockingProgress().addListener(listener);
+        Foo foo = mock(Foo.class);
+
+        //when
+        VerificationMode never = never();
+        verify(foo, never).doSomething("");
+
+        //then
+        assertThatHasBeenNotified(listener, foo, never, invocationWanted);
+    }
+
+    @Test
+    public void should_call_all_listeners_on_verify() {
+        //given
+        RememberingListener listener1 = new RememberingListener();
+        RememberingListener listener2 = new RememberingListener();
+
+        MockingProgress mockingProgress = ThreadSafeMockingProgress.mockingProgress();
+        mockingProgress.addListener(listener1);
+        mockingProgress.addListener(listener2);
+
+        Foo foo = mock(Foo.class);
+
+        //when
+        VerificationMode never = never();
+        verify(foo, never).doSomething("");
+
+        //then
+        assertThatHasBeenNotified(listener1, foo, never, invocationWanted);
+        assertThatHasBeenNotified(listener2, foo, never, invocationWanted);
+    }
+
+    @Test
+    public void should_not_call_listener_when_verify_was_called_incorectly() {
+        //given
+        RememberingListener listener = new RememberingListener();
+        MockingProgress mockingProgress = ThreadSafeMockingProgress.mockingProgress();
+
+        mockingProgress.addListener(listener);
+
+        //when
+        Foo foo = null;
+        try {
+            verify(foo).doSomething("");
+            fail("Exception expected.");
+        } catch (Exception e) {
+            //then
+            assertNull(listener.cause);
+        }
+    }
+
+    @Test
+    public void wrong_verify_type() {
+        //given
+        RememberingListener listener = new RememberingListener();
+        MockingProgress mockingProgress = ThreadSafeMockingProgress.mockingProgress();
+
+        mockingProgress.addListener(listener);
+
+        Foo foo = mock(Foo.class);
+        //when
+        try {
+            verify(foo).doSomething("");
+            fail("Exception expected.");
+        } catch (Throwable e) {
+            //then
+            assertThat(listener.cause).isInstanceOf(MockitoAssertionError.class);
+        }
+    }
+
+    private void assertThatHasBeenNotified(RememberingListener listener, Object mock, VerificationMode mode, String invocationWanted) {
+        assertThat(listener.mock).isEqualTo(mock);
+        assertThat(listener.mode).isEqualTo(mode);
+        assertThat(listener.data.getWanted().getMethod().getName()).isEqualTo(invocationWanted);
```

Comment:
@bric3 Yes I agree that LOD is more target to production...

I like your ideas for improving the assertions. I think this would improve the tests a lot.